src/test/java/org/jitsi/meet/test/base/AbstractParticipantHelper.java

```diff
@@ -0,0 +1,177 @@
+ */
+package org.jitsi.meet.test.base;
+
+import org.jitsi.meet.test.util.*;
+import org.openqa.selenium.*;
+
+public abstract class AbstractParticipantHelper
+{
+    /**
+     * The current room name used.
+     */
+    private String currentRoomName;
+
+    /**
+     * The conference first participant(owner) in the tests.
+     */
+    protected Participant<WebDriver> participant1;
+
+    /**
+     * The second participant.
+     */
+    protected Participant<WebDriver> participant2;
+
+    /**
+     * The third participant.
+     */
+    protected Participant<WebDriver> participant3;
+
+    /**
+     * Starts the owner and the seconds participant, if they are not started,
+     * and stops the third participant, if it is not stopped.
+     * participants(owner and 'second participant').
+     */
+    public void ensureTwoParticipants()
+    {
+        waitForParticipant1ToJoinMUC();
+        waitForSecondParticipantToConnect();
+
+        // TODO do we need this???
+        //closeThirdParticipant();
+    }
+
+    /**
+     * Waits until the owner joins the room, creating and starting the owner
+     * if it hasn't been started.
+     */
+    public void waitForParticipant1ToJoinMUC()
+    {
+        if (participant1 == null)
+        {
+            participant1
+                = ParticipantFactory.getInstance()
+                    .createParticipant("web.participant1");
+        }
+
+        if (participant1.isHungUp())
+        {
+            // join room
+            participant1.joinConference(currentRoomName);
+        }
+
+        MeetUtils.waitForParticipantToJoinMUC(participant1.getDriver(), 15);
+    }
+
+    /**
+     * Setup helper by generating the room name to use.
+     */
+    public void setup()
+    {
+        this.currentRoomName
+            = "torture" + String.valueOf((int)(Math.random()*1000000));
+    }
+
+    /**
+     * Cleans up by quiting all participants.
+     */
+    public void cleanup()
+    {
+        quitParticipant(participant1);
+        quitParticipant(participant2);
+        quitParticipant(participant3);
+    }
+
+    /**
+     * Quits a participant if initialized.
+     * @param participant the participant to quit.
+     */
+    private void quitParticipant(Participant<WebDriver> participant)
+    {
+        if (participant == null)
+            return;
+
+        participant.quit();
+    }
+
+    /**
+     * Waits until {@code secondParticipant} has joined the conference (its ICE
+     * connection has completed and has it has sent and received data).
+     */
+    public void waitForSecondParticipantToConnect()
+    {
+        if (participant2 == null)
+        {
+            participant2 = ParticipantFactory.getInstance()
+                .createParticipant("web.participant2");
+        }
+
+        if (participant2.isHungUp())
+        {
+            participant2.joinConference(currentRoomName);
+        }
+
+        MeetUtils.waitForParticipantToJoinMUC(participant2.getDriver(), 10);
```

Comment:
All those things should be part of participant.

Reply:
Moved.